shell/platform/android/external_view_embedder/external_view_embedder.cc

```diff
@@ -9,32 +9,47 @@
 namespace flutter {
 
 AndroidExternalViewEmbedder::AndroidExternalViewEmbedder(
-    std::shared_ptr<PlatformViewAndroidJNI> jni_facade)
-    : ExternalViewEmbedder(), jni_facade_(jni_facade) {}
+    std::shared_ptr<AndroidContext> android_context,
+    std::shared_ptr<PlatformViewAndroidJNI> jni_facade,
+    const AndroidSurface::Factory& surface_factory)
+    : ExternalViewEmbedder(),
+      android_context_(std::move(android_context)),
+      jni_facade_(std::move(jni_facade)),
+      surface_factory_(surface_factory),
+      surface_pool_(std::make_unique<SurfacePool>()) {}
 
 // |ExternalViewEmbedder|
 void AndroidExternalViewEmbedder::PrerollCompositeEmbeddedView(
     int view_id,
     std::unique_ptr<EmbeddedViewParams> params) {
-  // TODO(egarciad): Implement hybrid composition.
-  // https://github.com/flutter/flutter/issues/55270
   TRACE_EVENT0("flutter",
                "AndroidExternalViewEmbedder::PrerollCompositeEmbeddedView");
+
+  auto rtree_factory = RTreeFactory();
+  view_rtrees_[view_id] = rtree_factory.getInstance();
   picture_recorders_[view_id] = std::make_unique<SkPictureRecorder>();
-  picture_recorders_[view_id]->beginRecording(SkRect::Make(frame_size_));
+  picture_recorders_[view_id]->beginRecording(SkRect::Make(frame_size_),
+                                              &rtree_factory);
 
   composition_order_.push_back(view_id);
+  // Update params if they changed.
+  if (view_params_.count(view_id) == 1 &&
+      view_params_[view_id] == *params.get()) {
+    return;
+  }
+  view_params_[view_id] = EmbeddedViewParams(*params.get());
 }
 
 // |ExternalViewEmbedder|
 SkCanvas* AndroidExternalViewEmbedder::CompositeEmbeddedView(int view_id) {
-  return picture_recorders_[view_id]->getRecordingCanvas();
+  if (picture_recorders_.count(view_id) == 1) {
```

Comment:
Nit: I think this is where you'll use `find` with an iterator deference. This technique performs map lookup twice. Not the end of the world, but unnecessary.

Reply:
I think you used the same technique right above as well.

Reply:
Acknowledged.